runtime/src/qio/qio.c

```diff
@@ -2890,6 +2890,7 @@ qioerr _qio_buffered_read(qio_channel_t* ch, void* ptr, ssize_t len, ssize_t* am
       }
       // Return early on an error or on EOF.
       if( err ) {
+        // do not update 'amt_read' because zero bytes were read on EOF
```

Comment:
a nit: amt_read was already update in the above `case` statements; it's just that we don't need to act on it.

Reply:
do you mean `num_read` was already updated in the case statements? (`amt_read` was updated in the previous iteration of the while loop)

Reply:
Oh, sorry, yes, you are right. Ignore my comment please!